### ISSUE TITLE
Setting 'padding' to a string will fail exporting Conv2d operator to ONNX.

### ISSUE THREAD

**Issue by miccio-dk** [pytorch/pytorch#68880]
## 🚀 Feature
Implement the operator `_convolution_mode` in the ONNX exporter 

## Motivation
Currently, the ONNX exporter fails when trying to export a model using convolutional layers (e.g. Conv2d) with padding="same":
> Exporting the operator _convolution_mode to ONNX opset version X is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.

## Pitch
I would like to be able to export my models using padded 2d convolution to ONNX

## Additional context
I'm using pytorch 1.10.0 from conda

**Comment by fatcat-z** [comment]
Could you please provide a simple example of showing how your code looks like?

**Comment by miccio-dk** [comment]
Thanks for reaching out.   
The message occurs with a model as simple as:
```py
dummy_data = torch.randn(1, 2, 64, 64)
m = torch.nn.Sequential(torch.nn.Conv2d(2, 32, 3, padding='same'), torch.nn.Conv2d(32, 16, 3, padding='same'))
# output shape should be [1, 16, 64, 64]
```

And this is my call to export:
```py
torch.onnx.export(
    model=m,
    args=dummy_data ,
    f='onnx_model_path/exported_model.onnx',
    verbose=False,
    export_params=True,
    do_constant_folding=False,
    input_names=['input'],
    #opset_version=11,  # same error with 
    output_names=['output'])
```

Hope this helps!

**Comment by fatcat-z** [comment]
This is being tracked internally by Microsoft at https://msdata.visualstudio.com/Vienna/_workitems/edit/1532364

**Comment by edgexspear** [comment]
> Thanks for reaching out. The message occurs with a model as simple as:
> 
> ```python
> dummy_data = torch.randn(1, 2, 64, 64)
> m = torch.nn.Sequential(torch.nn.Conv2d(2, 32, 3, padding='same'), torch.nn.Conv2d(32, 16, 3, padding='same'))
> # output shape should be [1, 16, 64, 64]
> ```
> 
> And this is my call to export:
> 
> ```python
> torch.onnx.export(
>     model=m,
>     args=dummy_data ,
>     f='onnx_model_path/exported_model.onnx',
>     verbose=False,
>     export_params=True,
>     do_constant_folding=False,
>     input_names=['input'],
>     #opset_version=11,  # same error with any other version <= 14
>     output_names=['output'])
> ```
> 
> Hope this helps!

Did you solve this problem please?

**Comment by SysuJayce** [comment]
Suffering from the same problem.

If we use `padding="same"` in Conv Module, training stage will be ~5% faster, but can not convert to onnx.

**Comment by Xavier-i** [comment]
Have the same problem here, any temp solution?

**Comment by ziggy1209** [comment]
same issue here.... looks like it's still not solved yet

**Comment by shubhambhokare1** [comment]
PR https://github.com/pytorch/pytorch/pull/89107 should resolve. Will close when this merged into master.
